book/08-customizing-git/sections/attributes.asc

```diff
@@ -142,20 +142,20 @@ index 88839c4..4afcb7c 100644
  Color Type                      : RGB with Alpha
 ----
 
-You can easily see that the file size and image dimensions have both changed.
+你一眼就能看出文件大小和图像尺寸发生了变化。
 
 [[_keyword_expansion]]
-==== Keyword Expansion
+==== 关键字扩展
 
 (((keyword expansion)))
-SVN- or CVS-style keyword expansion is often requested by developers used to those systems.
-The main problem with this in Git is that you can't modify a file with information about the commit after you've committed, because Git checksums the file first.
-However, you can inject text into a file when it's checked out and remove it again before it's added to a commit.
-Git attributes offers you two ways to do this.
+习惯了 SVN 和 CVS 的开发者经常希望能在 Git 中加入关键字扩展功能。
+在 Git 中实现这一功能的最大障碍在于，你无法在一个文件被提交后修改它，因为它的校验和在提交时已经算好了。
```

Comment:
@spacewander 持保留意见。

这个措辞可能需要改一下，但是我觉得准确来讲原文的意思是：

如果仅仅是修改文件，那么实际上是没有什么问题的。但是问题在于现在要做的是这个 commit 的唯一 ID，而在 Git 中，commit 利用 SHA 校验值来表示（SVN 是 revision 序号），确定该校验值的前提是确定文件内容，这样就构成了一个死链...

关键问题在于：修改文件所需要的信息，恰好是从文件本身的所有内容（包括被修改的部分），这里出现了 `commit` 这个词，仅仅是因为 Git 在 commit 时使用了这个校验而已。我认为这也是后半句 `because Git checksums the file first` 所要强调的，在文件被 checksum 之后再修改它，那么这个 checksum 就无效了。

另外，原文中出现了 `with`，但是后面没有承接的动词，所以我觉得这个 `with` 就是 `使用` 的意思。